samples/subsys/ipc/openamp_rsc_table/src/main_remote.c

```diff
@@ -130,10 +140,10 @@ static void new_service_cb(struct rpmsg_device *rdev, const char *name,
 
 int mailbox_notify(void *priv, uint32_t id)
 {
-	ARG_UNUSED(priv);
+	struct ipm_channel *ipm_ch = (struct ipm_channel *)priv;
 
 	LOG_DBG("%s: msg received", __func__);
-	ipm_send(ipm_handle, 0, id, NULL, 0);
+	ipm_send(ipm_handle, 0, ipm_ch->channel_id, NULL, 0);
```

Comment:
copy past from https://github.com/zephyrproject-rtos/zephyr/pull/78228#discussion_r1757154858:

This probably introduces a regression on existing platforms. I can see two assumptions in this patch:
1) The IPM channel ID used for RX and TX is the same. The initial implementation is that
[channel id 0 is used for RX](https://elixir.bootlin.com/zephyr/v3.7.0/source/lib/open-amp/resource_table.h#L20) and [channel id 1 is usedfor Tx](https://elixir.bootlin.com/zephyr/v3.7.0/source/lib/open-amp/resource_table.h#L21).
    
2) You assume that we receive a first message before sending the first one (ipm_ch->channel_id is set in platform_ipm_callback()), which seems like a dangerous assumption to me.

Some simple solutions could be to put the code under platform configuration or, perhaps more generically, to define `CONFIG_IPM_RX_CHANNEL_ID` and `CONFIG_IPM_TX_CHANNEL_ID` to configure the channel IDs.

Reply:
I think using the CONFIG_ definitions would work for us.